homeworks/08-functional-reactive-programming/src/App.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  combineLatest, from, interval, of, Subject,
+} from 'rxjs';
+import { useObservableState } from 'observable-hooks';
+import { map, scan } from 'rxjs/operators';
+import './App.css';
+
+import { Grid } from './components/Grid';
+
+import { generateGrid, getRandomTarget } from './utils';
+
+import { CellContent } from './types';
+
+export const App: React.FC = () => {
+  const [goodShots, setGoodShots] = useState<number>(0);
+
+  const shotTarget = new Subject<CellContent>();
+  const shotTarget$ = shotTarget.asObservable();
+
+  const targetIndex$ = interval(1000).pipe(map(() => getRandomTarget()));
+
+  const grid$ = of(generateGrid());
+
+  const cellsWithTarget$ = combineLatest([targetIndex$, grid$])
+    .pipe(
+      map(([targetIndex, grid]) => {
+        const newGrid = [...grid];
+        newGrid[targetIndex].content = CellContent.CAT;
+        return newGrid;
+      }),
+    );
+
+  const goodShots$ = from(shotTarget$)
+    .pipe(
+      map((target) => (target === CellContent.CAT ? 1 : 0)),
+      scan((acc, count) => acc + count, 0),
+    );
+
+  const cellsWithTarget = useObservableState(cellsWithTarget$, []);
+
+  useEffect(() => {
+    const sub = goodShots$.subscribe((v) => {
+      setGoodShots(v);
+    });
+    return sub.unsubscribe;
+  }, []);
```

Comment:
Why is dependency array empty?

Reply:
Using own hook i've removed this extra logic to use goodShots as state in app -> https://github.com/vbarbasins/evo-ts-bootcamp/pull/9/commits/0019021647f6172788cb12977ae64e8043b8ec28